pkg/commands/compute/language_go.go

```diff
@@ -123,6 +128,50 @@ func (g *Go) Dependencies() map[string]string {
 	return deps
 }
 
+// Imports returns all source code imported packages.
+func (g *Go) Imports() []string {
```

Comment:
How precise are you aiming this to be?  This wouldn't take into account unused vendoring, build tags, etc.  A `go list` invocation with the appropriate flags might give a more accurate reading here.  I'd think you'd also need to call it with `scripts.env_vars` set.

Reply:
This is just a _rough_ best effort. I don't think we need to be that specific with the details.

My understanding is that we're just trying to get a feel for popular frameworks that people are using. Which admittedly we'd get from the `Dependencies()` but I wanted to account for the difference between a popular package showing up as a non-direct dependency vs the user actually importing and _using_ that dependency explicitly.

The reason for choosing go.sum over go.mod was to try and account for that scenario where it might be useful to know if there's a popular package underlying a bunch of projects (even if they're not directly used).

Reply:
But as mentioned, I'll take a look at parsing `go.mod` instead to get both explicit and implicit dependencies.

I don't think there is the same for Rust/JavaScript unfortunately, so those will have to be manually parsed.

Reply:
> The reason for choosing go.sum over go.mod was to try and account for that scenario where it might be useful to know if there's a popular package underlying a bunch of projects (even if they're not directly used).

`go.sum` will be such a firehose without context that I don't think it'll give you useful data.  The output of `go mod graph` would be better because you'd have the context of what modules depend on other modules.  Combined with the data from `go.mod` you'd be able to filter out what's actually used vs. what's depended upon.

> But as mentioned, I'll take a look at parsing go.mod for Import() instead, to have an easier time parsing out the explicit dependencies rather than using the ast package.

Some confusion here: for `Dependencies()` I think you need to take `go.mod` into account, even if you are reading `go.sum` or running `go mod graph`.  For `Import()` I think what you have is ok, but `go list -f {{.Imports}}` with the appropriate platform settings will be more accurate.

Reply:
**UPDATE:** After some internal discussions, the following will be happening...

1. `Import()` will be removed completely (i.e. no parsing of source code files, and no `CLI_PackageInfoImported%d`).
2. `Dependencies()` for Go I'll going to switch to `go.mod`.
3.  The config.toml `[telemetry]` section will become `[wasm-metadata]`.
4. Environment variable `TelemetryDisable` will be renamed to align with '[wasm-metadata]' naming.
5. The `telemetry` command (from https://github.com/fastly/cli/pull/1013) will be moved to `compute metadata`.